functions/index.js

```diff
@@ -18,28 +19,82 @@ admin.initializeApp();
 
 
 
-const sections = ["AR", "CGC", "GC", "GM", "EL", "IN", "SV", "MA", "MT", "PH", "MX", "SIE", "SC"]
+const sections = ["AR", "CGC", "GC", "GM", "EL", "IN", "SV", "MA", "MT", "PH", "MX", "SIE", "SC", "NONE"]
 
+function set_error_callback (error) {
+    if (error) {
+      console.log('Data could not be saved.' + error)
+    } else {
+      console.log('Data saved successfully.')
+    }
+}
 
-exports.testScheduledFunction = functions.region('europe-west1').pubsub.schedule("15 * * * *").onRun((context) => {
+exports.resetVotesScheduledFunction = functions.region('europe-west1').pubsub.schedule("0 * * * *").onRun((context) => {
 
-    console.log("It's minute 15!");
+    console.log("It's minute 0!");
   
     const db = getDatabase();
-    const ref = db.ref('Zones');
+    const refZones = db.ref('Zones');
+    const refUsers = db.ref('Users');
 
 
-    ref.once('value').then( (snapshot) => {
+    refZones.once('value').then( (zonesSnapshot) => {
 
-        console.log("snapshot key " + snapshot.key)
+        // reset for each zone
+        zonesSnapshot.forEach((zoneSnapshot) => {
+
+            var zone_name = zoneSnapshot.key
+
+            zoneSnapshot.forEach((zoneChildSnapshot) => {
+                
+                if (zoneChildSnapshot.key != "owner"){
+
+                    var section_name = zoneChildSnapshot.key
+
+                    // reset count to 0
+                    refZones.child(zone_name).child(section_name).set(0, set_error_callback)
```

Comment:
Is there any way to ensure that it is the optimal way to do queries. I don't know if there is like a prefetching system to use? It seems like it is doing a query to get each child, where as we could probably do only one. (See [N+1 query problem](https://planetscale.com/blog/what-is-n+1-query-problem-and-how-to-solve-it))

Reply:
[Oh OK, it's good](https://stackoverflow.com/questions/37515203/how-to-structure-data-in-firebase-to-avoid-n1-selects). Seems like Firebase don't work like a traditional relational DB